src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -230,6 +230,28 @@ static ColumnWithTypeAndName readColumnWithTimestampData(std::shared_ptr<arrow::
     return {std::move(internal_column), std::move(internal_type), column_name};
 }
 
+static ColumnWithTypeAndName readColumnWithTimeData(std::shared_ptr<arrow::ChunkedArray> & arrow_column, const String & column_name)
+{
+    const auto & arrow_type = static_cast<const arrow::Time32Type &>(*(arrow_column->type()));
```

Comment:
Isn't it possible to corrupt data to have a wrong cast here? Is it validated by arrow library?

Reply:
Nice catch, I have separated it into two functions that use appropriate types. Can you enable CI?

Reply:
Oh, I meant a bit different case, but it's good that we found it :) When type tag is arrow::Type::TIME32, but actual type would be different from `arrow::Time32Type` is it possible to corrupt data in such way?

Also it's good to add tests-case to cover new code. I've added can-be-tested label

Reply:
Hm.. I am not exactly sure, tbh. What I can say is that there is already code that does exactly the same, see parquet timestamp conversion [example](https://github.com/ClickHouse/ClickHouse/blob/b26cc5a1ba791df03e56117c6207aec97b25b58b/src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp#L215).

I'll be adding tests in a few.

Reply:
I'll actually set this as draft while I don't have the tests

Reply:
Open for review again :)